x/cdp/keeper/cdp.go

```diff
@@ -410,6 +413,28 @@ func (k Keeper) CalculateCollateralToDebtRatio(ctx sdk.Context, collateral sdk.C
 	return collateralBaseUnits.Quo(debtTotal)
 }
 
+// LoadAugmentedCDP creates a new augmented CDP from an existing CDP
+func (k Keeper) LoadAugmentedCDP(ctx sdk.Context, cdp types.CDP) (types.AugmentedCDP, sdk.Error) {
+	// calculate collateralization ratio
+	collateralizationRatio, err := k.CalculateCollateralizationRatio(ctx, cdp.Collateral, cdp.Principal, cdp.AccumulatedFees)
```

Comment:
Technically this will give the collateralization ratio of the CDP at the point the fees were last updated, rather than what it is now. The difference will be small for low inflation rates, so we could just log an issue and fix it later.
Code for calculating the latest fees is used in deposit, withdraw, etc methods on the keeper.

Reply:
https://github.com/Kava-Labs/kava/pull/347/commits/8dcc94d5df79fbabdb40edc65817e9d707490cf5

Reply:
Since the CDP isn't updated after fee calculation, this change leads to the CDP showing a collateralization ratio which includes accumulatedFees, while accumulatedFees contains an empty array:

```- cdp:
- cdp:
    id: 2
    owner: kava1xy7hrjy9r0algz9w3gzm8u6mrpq97kwta747gj
    collateral:
    - denom: xrp
      amount: "150010000"
    principal:
    - denom: usdx
      amount: "10000000"
    accumulated_fees:
    - denom: usdx
      amount: "561"
    fees_updated: 2020-01-27T17:59:00.561629Z
  collateral_value:
    denom: usdx
    amount: "30002000"
  collateralization_ratio: "3.000031698221729761"
- cdp:
    id: 3
    owner: kava12jk3szk45afmvjc3xc6kvj4e40tuy2m8ckgs03
    collateral:
    - denom: xrp
      amount: "150000000"
    principal:
    - denom: usdx
      amount: "10000000"
    accumulated_fees: []
    fees_updated: 2020-01-26T22:03:50.894251Z
  collateral_value:
    denom: usdx
    amount: "30000000"
  collateralization_ratio: "2.999831409474787517"
```

 CDP #2 received an additional deposit that updated the accumulated fees, while CDP #3's accumulated fees haven't been updated yet the invisible fees are included in the collateralization ratio calculation.

Reply:
I guess that is a bit confusing. I'm ok with it for now though